src/org/infinity/resource/ResourceFactory.java

```diff
@@ -591,6 +594,53 @@ public static boolean saveResource(Resource resource, Component parent)
     }
   }
 
+  /**
+   * If {@code output} is not {@code null}, shows confirmation dialog for saving resource.
+   * If user accepts saving then resource will be saved if it implements {@link Writable}
+   *
+   * @param resource Resource that must be saved
+   * @param entry Entry that represents resource. Must not be {@code null}
+   * @param parent Component that will be parent for dialog window. Must not be {@code null}
+   *
+   * @throws HeadlessException if {@link GraphicsEnvironment#isHeadless} returns {@code true}
+   * @throws NullPointerException If any argument is {@code null}
+   * @throws Exception If save will be cancelled
+   */
+  public static void closeResource(Resource resource, ResourceEntry entry, JComponent parent) throws Exception {
+    final Path output;
+    if (entry instanceof BIFFResourceEntry) {
+      output = FileManager.query(Profile.getRootFolders(), Profile.getOverrideFolderName(), entry.toString());
+    } else {
+      output = entry.getActualPath();
```

Comment:
In 3 cases there is also a check that `entry` is file entry:
* https://github.com/Argent77/NearInfinity/blob/922dc9e42addddca0fae9d906cb224229baedb38/src/org/infinity/resource/graphics/BamResource.java#L147-L152
* https://github.com/Argent77/NearInfinity/blob/922dc9e42addddca0fae9d906cb224229baedb38/src/org/infinity/resource/graphics/PltResource.java#L221-L226
* https://github.com/Argent77/NearInfinity/blob/922dc9e42addddca0fae9d906cb224229baedb38/src/org/infinity/resource/other/UnknownResource.java#L98-L103

It differs from implementation in all other resources. It is expected? Whether it is necessary to lead everything to one look and if yes, that to what?


Also, `PlainTextResource` search only in game root, not in all roots, as all other resources:
https://github.com/Argent77/NearInfinity/blob/922dc9e42addddca0fae9d906cb224229baedb38/src/org/infinity/resource/text/PlainTextResource.java#L123-L128
It is expected? Now I changed this to search on all roots

Reply:
**BamResource.java:** Extra check for FileResourceEntry is not strictly needed for this and the two other instances; just a side effect of implementing similar routines at different times. Can be replaced by `ResourceFactory.closeResource()`.

**PlainTextResource.java:** Not intended. It can be changed search all roots.